app/scripts/components/discoveries/hub/index.tsx

```diff
@@ -1,28 +1,159 @@
+import {
+  LayoutProps,
+  useSlidingStickyHeaderProps
+} from '$components/common/layout-root';
 import PageHero from '$components/common/page-hero';
-import { Fold, FoldHeader, FoldTitle } from '$components/common/fold';
-import { Card, CardList } from '$components/common/card';
+import {
+  Fold,
+  FoldHeader,
+  FoldHeadline,
+  FoldTitle
+} from '$components/common/fold';
+import {
+  Card,
+  CardList,
+  CardMeta,
+  CardTopicsList
+} from '$components/common/card';
 import EmptyHub from '$components/common/empty-hub';
-
 import { PageMainContent } from '$styles/page';
 import { DISCOVERIES_PATH, getDiscoveryPath } from '$utils/routes';
+import TextHighlight from '$components/common/text-highlight';
+import Pluralize from '$utils/pluralize';
+import { Pill } from '$styles/pill';
+
+const allDiscoveries = Object.values(discoveries).map((d) => d!.data);
+
+const DiscoveryCount = styled(Subtitle)`
+  grid-column: 1 / -1;
+  display: flex;
+  gap: ${glsp(0.5)};
+
+  span {
+    text-transform: uppercase;
+    line-height: 1.5rem;
+  }
+`;
+
+const BrowseHeader = styled(FoldHeader)`
+  ${media.largeUp`
+    ${FoldHeadline} {
+      align-self: flex-start;
+    }
+
+    ${BrowseControls} {
+      padding-top: 1rem;
+    }
+  `}
+`;
+
+const topicsOptions = [
+  optionAll,
+  // TODO: human readable values for Taxonomies
+  ...Array.from(new Set(allDiscoveries.flatMap((d) => d.thematics || []))).map(
+    (t) => ({
+      id: t,
+      name: t
+    })
+  )
+];
+
+const sourcesOptions = [optionAll];
+
+const sortOptions = [
+  { id: 'name', name: 'Name' },
+  { id: 'pubDate', name: 'Date' }
+];
+
+const prepareDiscoveries = (data: DiscoveryData[], options) => {
+  const { sortField, sortDir, search, topic, source } = options;
+
+  let filtered = [...data];
+
+  // Does the free text search appear in specific fields?
+  if (search.length >= 3) {
+    const searchLower = search.toLowerCase();
+    filtered = filtered.filter(
+      (d) =>
+        d.name.toLowerCase().includes(searchLower) ||
+        d.description.toLowerCase().includes(searchLower) ||
+        d.thematics?.some((t) => t.toLowerCase().includes(searchLower))
+    );
+  }
+
+  if (topic !== optionAll.id) {
+    filtered = filtered.filter((d) => d.thematics?.includes(topic));
+  }
+
+  if (source !== optionAll.id) {
+    // TODO: Filter source
+  }
 
-/* eslint-disable-next-line fp/no-mutating-methods */
-const allDiscoveries = Object.values(discoveries)
-  .map((d) => d!.data)
-  .sort((a, b) => {
-    const getTime = (d: string) => {
-      const millis = new Date(d).getTime();
-      return isNaN(millis) ? -Infinity : millis;
-    };
+  /* eslint-disable-next-line fp/no-mutating-methods */
+  filtered.sort((a, b) => {
+    if (!a[sortField]) return Infinity;
 
-    return getTime(b.pubDate) - getTime(a.pubDate);
+    return a[sortField]?.localeCompare(b[sortField]);
   });
 
+  if (sortDir === 'desc' || sortField === 'pubDate') {
```

Comment:
Not sure I understand the `sortField === 'pubDate'` part. This makes sorting by date in ascending order impossible.

Reply:
Great catch! :D